language/move-lang/tests/move_check/typing/invalid_type_acquire.exp

```diff
@@ -0,0 +1,481 @@
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:14:9 ───
+    │
+ 14 │         move_to_sender(R { v })
+    │         ^^^^^^^^^^^^^^^^^^^^^^^ Invalid call to move_to_sender.
+    ·
+ 14 │         move_to_sender(R { v })
+    │                        ------- The resource cannot take any type arguments, but found 1 type arguments
+    │
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:18:9 ───
+    │
+ 18 │         T,
+    │         ^ Invalid 'acquires' annotation
+    ·
+ 18 │         T,
+    │         - Expected a nominal resource. Found the type: T
+    │
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:19:9 ───
+    │
+ 19 │         u64,
+    │         ^^^ Invalid 'acquires' annotation
+    ·
+ 19 │         u64,
+    │         --- Expected a nominal resource. Found the type: u64
+    │
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:20:9 ───
+    │
+ 20 │         X.R,
+    │         ^^^ Invalid 'acquires' annotation
+    ·
+ 20 │         X.R,
+    │         --- The type '0x1.X.R' was not declared in the current module. Global storage access is internal to the module'
+    │
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:10:12 ───
+    │
+ 21 │         S,
+    │         ^ Invalid 'acquires' annotation
+    ·
+ 21 │         S,
+    │         - Expected a nominal resource. Found the type: 0x1.M.S
+    ·
+ 10 │     struct S {}
+    │            - Declared as a normal struct here
+    │
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:22:9 ───
+    │
+ 22 │         R<u64>,
+    │         ^^^^^^ Invalid 'acquires' annotation
+    ·
+ 22 │         R<u64>,
+    │         ------ The resource cannot take any type arguments, but found 1 type arguments
+    │
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:23:9 ───
+    │
+ 23 │         R<T>
+    │         ^^^^ Invalid 'acquires' annotation
+    ·
+ 23 │         R<T>
+    │         ---- The resource cannot take any type arguments, but found 1 type arguments
+    │
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:32:9 ───
+    │
+ 32 │         destroy(move_from(a));
+    │         ^^^^^^^^^^^^^^^^^^^^^ Could not infer this type. Try adding an annotation
+    │
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:32:17 ───
+    │
+ 32 │         destroy(move_from(a));
+    │                 ^^^^^^^^^ Could not infer this type. Try adding an annotation
+    │
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:33:17 ───
+    │
+ 33 │         destroy(move_from<T>(a));
+    │                 ^^^^^^^^^^^^^^^ Invalid call to move_from.
+    ·
+ 33 │         destroy(move_from<T>(a));
+    │                           - Expected a nominal resource. Found the type: T
```

Comment:
Same question here as in the acquires case. It might be helpful (if possible) if the compiler could give the kind of the type at this point.